kardinal-cli/kubernetes/kubernetes_client.go

```diff
@@ -198,6 +229,47 @@ func (client *KubernetesClient) GetNamespacesByLabels(ctx context.Context, names
 	return &namespacesNotMarkedForDeletionnamespaceList, nil
 }
 
+// TODO merge kardinal-cli.kubernetes-client with kardinal-manager.cluster-manager, this method is in both objects
+func (client *KubernetesClient) EnsureNamespace(ctx context.Context, name string) error {
+	if name == istioSystemNamespace {
```

Comment:
Not sure why we need this? What happens if the user install Istio into different namespace?

Reply:
Also, namespace used by resources created by unmanaged resources should be responsibility of the user, no? So if I have a secret resource on namespace `my-space`, it also my responsibility to create the namespace `my-space` either before or in the same manifest file

Reply:
@lostbean yeap, makes sense.
I added this feat because the deployment was failing when the baseline namespace had not been created when deploying the ServiceAccount inside it but yes probably the namespaces should be created by the user.
I'll remove it